calcom/cal.com
# feat: multi round robin hosts

## What does this PR do?

Adds the possibility to select several round-robin-hosts for the respective event-type.

Fixes # [11953](https://github.com/calcom/cal.com/issues/11953)

Crucial changes which were necessary:

**1. Frontend:** 
Adds an input-field to the Event-Team-Tab by which the number of round-robin hosts which need to participate
in the event can be specified

**2. Availability:**
For multiple round-robin-hosts the way how the collective availability is determined needs to be changed.

**3. Select the correct number of hosts in the booking-handler**  
 
<img width="1195" alt="Cal_com" src="https://github.com/calcom/cal.com/assets/145436092/a6e423ed-a5a4-4f8e-9ffa-f3f3d37ceb0f">


## Type of change




## How should this be tested?

1. Create a Round-Robin Event-Type 
2. Under "Assignment" select some users which have a common availability
3. Provide the number of users which you want to particiate in a Round-Robin fashion
4. Now when trying to book the event, the datepicker should respect the new constraints
5. The hosts selection should now select the number of hosts which you provided with Round-Robin

A test is provided in the PR.
